great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -816,6 +853,25 @@ def run_profiler(
 
         return result
 
+    @staticmethod
+    def run_profiler_on_data(
+        data_context: "DataContext",  # noqa: F821
+        profiler_store: ProfilerStore,
+        batch_request: Union[dict, BatchRequest, RuntimeBatchRequest],
```

Comment:
@cdkini Since `batch_request` is required here, per your earlier point, we can thus make it required in the caller.  Thanks.